### PR TITLE
Feature/apidoc

### PR DESCRIPTION
**Pull request by jmcarp**
Replacement for #571.

### CODE REVIEW

File: webservices/resources/candidates.py [hunk 1]
```diff
@@ -1,290 +1,131 @@
...
+
+
+filter_fields = {
+    'candidate_id',
+    'candidate_status',
+    'district',
+    'incumbent_challenge',
+    'office',
+    'party',
+    'state',
```
Comment (LindsayYoung) [inline webservices/resources/candidates.py]: We need a year filter for `/candidates` is this list just for detail and history?

Comment (jmcarp) [inline webservices/resources/candidates.py]: This logic is for generic filtering, meaning we want an exact match between the query parameter and the column in the database. Year filtering is more complicated and varies more across endpoints, so I left it in the `get_candidate` and `get_candidates` methods. For example, https://github.com/jmcarp/openFEC/blob/feature/apidoc/webservices/resources/candidates.py#L59-L64

Comment (LindsayYoung) [inline webservices/resources/candidates.py]: Ok, that makes sense. Thanks for explaining!

**Comment by LindsayYoung** [comment]
This looks good to me.

File: webservices/rest.py [hunk 2]
```diff
@@ -4,23 +4,37 @@
...
+import smore.apispec
+from flask import abort
+from flask import request
+from flask import jsonify
+from flask import url_for
+from flask import render_template
+from flask import Flask
+from flask import Blueprint
 
 from flask import Flask, Blueprint, abort, request
```
Comment (arowla) [inline webservices/rest.py]: There are some duplicated imports here.

Comment (jmcarp) [inline webservices/rest.py]: Merge conflicts :confounded:. Fixing now.

**Pull request merged**